test/unit/transaction.test.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+/* jshint -W030 */
+var chai = require('chai')
+, expect = chai.expect
+, sinon = require('sinon')
+, Support = require(__dirname + '/support')
+, Sequelize = Support.Sequelize
+, dialect = Support.getTestDialect()
+, current = Support.sequelize
+, Promise = Sequelize.Promise;
+
+describe('Transaction', function() {
+
+  before(function () {
+    this.stub = sinon.stub(current, 'query').returns(Promise.resolve({}));
+
+    this.stubConnection = sinon.stub(current.connectionManager, 'getConnection')
+    .returns(Promise.resolve({ uuid: 'ssfdjd-434fd-43dfg23-2d', close : function() { }}));
+  });
+
+  beforeEach(function () {
+    this.stub.reset();
+    this.stubConnection.reset();
+  });
+
+  after(function () {
+    this.stub.restore();
+    this.stubConnection.restore();
+  });
+
+  it('should run auto commit query only when needed', function() {
+    var expectations = {
+      all: [
+        'START TRANSACTION;'
+      ],
+      mysql: [
+        'START TRANSACTION;',
+        'SET autocommit = 1;'
```

Comment:
Hi @janmeier @mickhansen 

I have been investigating the reason why MySQL required `autocommit`. @janmeier is correct, as per `MySQL` docs we don't need to set it for `START TRANSACTION` which implies autocommit is set to false. 

But there is a bug in sequelize as noted in #3568, connection which use the `autocommit off` once will remain `off` until user manually doesn't switch it back `on`. This should happen inside `commit()` and `rollback()` automatically?

So can we merge this PR with MySQL `autocommit` set to 1 by default, as its supposed to? Then I will work on #3568 later on and will send a PR to fix it.